delivery_free_fee_removal/models/sale_order.py

```diff
@@ -11,9 +11,11 @@ def action_confirm(self):
         res = super().action_confirm()
         # Invoice all the free delivery line on order confirmation
         # Or the order will never be fully invoiced.
-        delivery_lines = self.order_line.filtered(
-            lambda line: line.order_id.state == "sale" and line.is_free_delivery
-        )
-        for line in delivery_lines:
-            line.qty_delivered = line.qty_invoiced = line.product_uom_qty
+        for order in self:
```

Comment:
Was the former code problematic ?

Reply:
not problem but method `action_confirm` maybe call many sale_orders so I add loop order record

Reply:
Yes, but from >v16, all relation fields can be joined together.

So, you can write `self.order_line.order_id.partner_id` without problems. It will return all the partners as a recordset.

Reply:
So it not necessary loop order record, but I sill use filtered order state direly?

Reply:
Yes you can as it removes a carriage return for readability

Reply:
I think former code is fine and more maintainable than my code so I revert this changes